### PR TITLE
Trying to make GetSpiralGenerator utility for the issue #5197

### PR DESCRIPTION
**Pull request by IcarusRTS**
There is only the most recent problem to tackle from the previous pull request that has been accidentally closed while I was looking in the mobile.

### CODE REVIEW

File: LuaUI/Widgets/test1_sg.lua [hunk 1]
```diff
@@ -0,0 +1,14 @@
+VFS.Include("LuaRules/Utilities/GetSpiralGenerator.lua")
+
+local spiralWithExtraParameters = Spring.Utilities.GetSpiralGenerator(12, 34, 3, "n", false)
```
Comment (sprunk) [inline LuaUI/Widgets/test1_sg.lua:3]: The test specified in #5197 looks different.
```suggestion
local spiralWithExtraParameters = Spring.Utilities.GetSpiralGenerator(12, 34, { step = 3, startingDirection = "n",  clockwise = false })
```

Comment (IcarusRTS) [reply LuaUI/Widgets/test1_sg.lua:3]: In my defense, I thought these (aka { and }) was causing a syntax error when i tried to put it into the start_unit_setup gadget before I reverted it back to normal and before I found out that test widget weren't even properly loaded. I will just revert it back and then test it, give me a moment.

Comment (sprunk) [reply LuaUI/Widgets/test1_sg.lua]: Perhaps look around other Lua widgets and check how various functions manage to pass and accept { tables }.

**Comment by IcarusRTS** [comment]
[t=00:01:31.836375][f=-000001] Failed to load: test1_sg.lua, [string "LuaRules/Utilities/GetSpiralGenerator.lua"]:41: attempt to perform arithmetic on local 'step' (a table value)
[t=00:01:31.836980][f=-000001] Failed to load: test2_sg.lua, [string "LuaRules/Utilities/GetSpiralGenerator.lua"]:9: attempt to perform arithmetic on local 'x' (a nil value)

These are the current issues right now, i will see if i can fix them.

**Comment by IcarusRTS** [comment]
> [t=00:01:31.836375][f=-000001] Failed to load: test1_sg.lua, [string "LuaRules/Utilities/GetSpiralGenerator.lua"]:41: attempt to perform arithmetic on local 'step' (a table value) [t=00:01:31.836980][f=-000001] Failed to load: test2_sg.lua, [string "LuaRules/Utilities/GetSpiralGenerator.lua"]:9: attempt to perform arithmetic on local 'x' (a nil value)
> 
> These are the current issues right now, i will see if i can fix them.

Yeah... I dunno how to fix these right now.

**Comment by sprunk** [comment]
That's fine, it can wait until you get better at lua.

**Pull request closed without merging**